MovieQuiz/Services/MoviesLoader.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  MoviesLoader.swift
+//  MovieQuiz
+//
+//  Created by Кирилл Брызгунов on 25.11.2022.
+//
+
+import Foundation
+
+protocol MoviesLoading {
+    func loadMovies(headler: @escaping (Result<MostPopularMovies, Error>) -> Void)
+}
+
+
+struct MoviesLoader: MoviesLoading {
+    // MARK: - NetworkClient
+    private let networkClient = NetworkClient()
+    
+    // MARK: - URL
+    private var mostPopularMoviesUrl: URL {
+        guard let url = URL(string: "https://imdb-api.com/en/API/Top250Movies/k_adn67lp5") else {
+            preconditionFailure("Unable to construct mostPopularMoviesUrl")
+        }
+        return url
+    }
+    
+    func loadMovies(headler: @escaping (Result<MostPopularMovies, Error>) -> Void) {
+        networkClient.fetch(url: mostPopularMoviesUrl) { result in
+            switch result {
+            case .failure(let error):
+                headler(.failure(error))
+            case .success(let data):
+                guard let jsonData = try? JSONDecoder().decode( MostPopularMovies.self, from: data ) else { return }
```

Comment:
Предположим, что API  сервиса IMDB поменялось, например вместо errorMessage поле стало приходить как error (в обратную сторону тоже верно, можешь попробовать поменять у себя в структуре). Тогда декодер перестанет правильно распознавать структуру и результат ошибки вызывающий код никогда не получит. Значит интерфейс будет крутить бесконечно активити индикатор и приложение возможно будет выглядеть сломанным. 

Поэтому либо в guard-else вернуть ошибку, либо использовать ошибку словленную через do-catch
```suggestion
                guard let jsonData = try? JSONDecoder().decode(MostPopularMovies.self, from: data) else { 
                    // Вернуть ошибку в handler 
                    return
                }
```

Reply:
это место критичное в этой работе — обязательно исправить

Reply:
@uglyqurill можно прямо здесь задавать вопрос и тегнуть меня через @, не обязательно оставлять комменты в коде)

Этот код через guard-else можно написать так:

1) Завести свою ошибку. Часто пишут новый тип enum, например MovieLoaderError. Расширяют его типом Error. 
Вот так: enum MovieLoaderError: Error. И описывают варианты ошибок. Например здесь wrongApiResponse 

2) так как новый тип удовлетворяет Error, его можно вернут в блоке else как `handler(.failure(MovieLoaderError. wrongApiResponse))`

Reply:
@ABashkirova Понял, спасибо!
Хорошо, буду тут тегать) Я пока не такой опытный пользователь гитхаба вхахвах, оставлял комментарии как мог, надеюсь это было не сильно неудобно

Reply:
Ничего, главное разобрались :) с гитхабом еще подружишься)